### ISSUE TITLE
Postgres:  Fixes for LT02 not applied, as it would result in an unparsable file. Problem with hyphen symbol (-) in dbt sources

### ISSUE THREAD

**Issue by FelipeNicolodi** [sqlfluff/sqlfluff#5425]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I'm trying to use sqlfluff in my dbt project, and it's returning the error "Fixes for LT02 not applied, as it would result in an unparsable file. Please report this as a bug with a minimal query which demonstrates this warning" when I try to format any code that contains a {{source}} with hyphen (-). For example: `{{ source('app_integrate-module','ptc_request_step') }}`

If I change the hyphen (-) to an underscore (_) - that is, `{{ source('app_integrate_module','ptc_request_step') }}` - the formatter works.
If I test it with the way dbt compiles this source - that is, `"postgres"."app_integrate-module"."ptc_request_step"` - the formatter also works.

So, the problem occurs only when the hyphen (-) is in the {{source}}.

### Expected Behaviour

No parse error.

### Observed Behaviour

Described above.

### How to reproduce

Query example:

```
SELECT prs.request_id,
    prs.created_at,
    prs.updated_at
   FROM {{ source("app_integrate-module","ptc_request_step") }} prs
  WHERE NOT prs.is_deleted
```

Result:
`WARNING    Fixes for LT02 not applied, as it would result in an unparsable file. Please report this as a bug with a minimal query which demonstrates this warning. `

### Dialect

postgres

### Version

sqlfluff: 2.3.5
python: 3.10.11
dbt: 1.6.0

### Configuration

[sqlfluff]
templater = jinja
dialect = postgres
exclude_rules = AL04, AL05, AL06, AL07, AM01, AM02, AM03, AM04, AM05, AM07, 
                CV01, CV02, CV03, CV04, CV05, CV06, CV07, CV08, CV09, CV10, 
                CV11, LT03, LT05, LT07, LT08, LT12, LT13, RF01, RF02, RF03,
                RF04, RF05, RF06, ST01, ST02, ST03, ST06, ST07, ST08, TQ01 
ignore = templating  
large_file_skip_byte_limit = 0 
max_line_length = 0  

[sqlfluff:layout:type:alias_expression]
spacing_before = align
align_within = select_clause
spacing_after = touch

[sqlfluff:indentation]
tab_space_size = 2
indent_unit = space
indented_joins = false
indented_using_on = true
allow_implicit_indents = true
indented_on_contents = false
indented_ctes = false

[sqlfluff:rules:aliasing.table]
aliasing.table = explicit

[sqlfluff:rules:aliasing.column]
aliasing.column = explicit

[sqlfluff:rules:aliasing.expression]
allow_scalar = True

[sqlfluff:rules:ambiguous.column_references]     
group_by_and_order_by_style = consistent

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.identifiers]
extended_capitalisation_policy = lower
unquoted_identifiers_policy = all

[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = upper

[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = upper

[sqlfluff:rules:jinja.padding]
single_space = true

[sqlfluff:rules:layout.spacing]
no_trailing_whitespace = true
extra_whitespace = false

[sqlfluff:layout:type:comma]
line_position = leading

[sqlfluff:rules:layout.functions]
no_space_after_function_name = true

[sqlfluff:rules:layout.select_targets]
wildcard_policy = multiple

[sqlfluff:rules:layout.set_operators]
set_operator_on_new_line = ['UNION', 'UNION ALL']

[sqlfluff:rules:structure.nested_case]

[sqlfluff:rules:structure.subquery]
forbid_subquery_in = join

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by 0xBigBoss** [comment]
I am too seeing a similar warning on postgres.

```shell
sqlfluff format  --config supabase/sqlfluff.cfg supabase/migrations/20240521012932_create_activity_feed_function.sql        

==== finding fixable violations ====
WARNING    Fixes for LT01 not applied, as it would result in an unparsable file. Please report this as a bug with a minimal query which demonstrates this warning.                                                                                                                                         
WARNING    Fixes for LT02 not applied, as it would result in an unparsable file. Please report this as a bug with a minimal query which demonstrates this warning.                                                                                                                                         
WARNING    Fixes for LT01 not applied, as it would result in an unparsable file. Please report this as a bug with a minimal query which demonstrates this warning.                                                                                                                                         
WARNING    Fixes for LT02 not applied, as it would result in an unparsable file. Please report this as a bug with a minimal query which demonstrates this warning.                                                                                                                                         
WARNING    Fixes for LT01 not applied, as it would result in an unparsable file. Please report this as a bug with a minimal query which demonstrates this warning.                                                                                                                                         
WARNING    Fixes for LT02 not applied, as it would result in an unparsable file. Please report this as a bug with a minimal query which demonstrates this warning.    
```

my sql

```sql
create or replace view activity_feed with (security_barrier = on) as
select a.created_at                  as created_at,
       a.event_name                  as event_name,
       case when a.from_user_id = from_p.id then (case when a.from_user_id = ( select auth.uid() )
                                                           then ( select auth.uid() ) end,
                                                  from_p.name,
                                                  from_p.avatar_url,
                                                  from_p.send_id,
                                                  ( select array_agg(name)
                                                    from tags
                                                    where user_id = from_p.id and status = 'confirmed' )
           )::activity_feed_user end as from_user,
       case when a.to_user_id = to_p.id then (case when a.to_user_id = ( select auth.uid() )
                                                       then ( select auth.uid() ) end,
                                              to_p.name,
                                              to_p.avatar_url,
                                              to_p.send_id,
                                              ( select array_agg(name)
                                                from tags
                                                where user_id = to_p.id and status = 'confirmed' )
           )::activity_feed_user end as to_user,
       a.data                        as data
from activity a
         left join profiles from_p on a.from_user_id = from_p.id
         left join profiles to_p on a.to_user_id = to_p.id
where a.from_user_id = ( select auth.uid() )
   or a.to_user_id = ( select auth.uid() )
group by a.created_at, a.event_name, a.from_user_id, a.to_user_id, from_p.id, from_p.name, from_p.avatar_url,
         from_p.send_id, to_p.id, to_p.name, to_p.avatar_url, to_p.send_id, a.data;
```
